src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.spec.ts

```diff
@@ -508,4 +523,34 @@ describe('IscsiTargetFormComponent', () => {
       });
     });
   });
+
+  function validate_auth(formHelper: FormHelper) {
+    formHelper.expectErrorChange('auth.user', 'short', 'pattern');
+    formHelper.expectValidChange('auth.user', 'thisIsCorrect');
+    formHelper.expectErrorChange('auth.user', '##?badChars?##', 'pattern');
+    formHelper.expectErrorChange(
+      'auth.user',
+      'thisUsernameIsWayyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyTooBig',
+      'pattern'
+    );
```

Comment:
How about the following:
```suggestion
    const expectValidAuthField = (fieldName: string) => {
      formHelper.expectErrorChange(fieldName, 'short', 'pattern');
      formHelper.expectValidChange(fieldName, 'thisIsCorrect');
      formHelper.expectErrorChange(fieldName', '##?badChars?##', 'pattern');
      formHelper.expectErrorChange(
        fieldName,
        'thisUsernameIsWayyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyTooBig',
        'pattern'
      );
    }
    expectValidAuthField('auth.user')
    expectValidAuthField('auth.password')
    expectValidAuthField('auth.mutual_user')
    expectValidAuthField('auth.mutual_password')
```

Reply:
Validation for user and password are different, so I had to create 2 separate methods.

Reply:
The only difference I can spot is that the to big pattern is longer for usernames, but if we would use that for the password fields should it show the same error if I'm not mistaken.

Anyway nice improvements ;)